.github/workflows/generalized-deployments-reproducible.yaml

```diff
@@ -0,0 +1,23 @@
+on: push
```

Comment:
It looks like this maps the default branch to `dev` and builds all other heads as-named. Do we want restrictions here to avoid cluttering the image registry?, e.g.
```yaml
on:
  push:
    branches:
      - main
    tags:
      - v*
```